linear-regression/helper_functions.py

```diff
@@ -0,0 +1,375 @@
+import pandas as pd
+from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score
+from sklearn.preprocessing import PowerTransformer
+from typing import Tuple
+import numpy as np
+import pytest
+
+
+def create_equatorial_distance_feature(df) -> pd.DataFrame:
+    """
+    Create a new feature 'distance_from_equator' based on the absolute value of 'Latitude' and bin the values into 10 bins.
+
+    Args:
+        df: pandas DataFrame - Input DataFrame containing 'Latitude' column.
+
+    Returns:
+        pandas DataFrame - DataFrame with the new feature 'distance_from_equator' and binned values in 'distance_from_equator_bin'.
+    """
+
+    # Create a new feature 'distance_from_equator' by taking the absolute value of 'Latitude'
+
+    #sales['Age Ranges'] = pd.cut(sales['Contact Age'], bins = range(20, 61, 5), right = False)
+    #sales['Age Ranges']
+    df1 = df.copy()
+    df1['distance_from_equator'] = df1['Latitude'].abs()
+    df1['distance_from_equator_bin'] = pd.cut(df1['distance_from_equator'], bins = 10, labels = False)
+    return df1
+
+
+def evaluate_model(y_true, y_pred):
+    """
+    Calculate Mean Squared Error (MSE), Mean Absolute Error (MAE), and R-squared (R2) score between true and predicted values.
+
+    Args:
+        y_true: Array-like of shape (n_samples,) - Ground truth target values.
+        y_pred: Array-like of shape (n_samples,) - Predicted target values.
+
+    Returns:
+        mse: float - Mean Squared Error.
+        mae: float - Mean Absolute Error.
+        r2: float - R-squared score.
+
+    Examples:
+        mse, mae, r2 = evaluate_model(y_true, y_pred)
+    """
+    # create the mse, mae, and r2 variables
+    n = len(y_true)
+    y_bar = (sum(y_true)) / n
+
+    # Calculate MSE
+    mse = (1 / n)*sum((y_true - y_pred)**2)
+    # Calculate MAE
+    mae = (1 / n)*sum(abs(y_pred - y_true))
+
+    # Calculate r-squared
+    r2 = sum((y_true - y_pred)**2)/sum((y_true - y_bar)**2)
```

Comment:
This is SSE/SST whereas R2 = SSR/SST. Do you remember how you can get R2 in terms of SSE?